### PR TITLE
[MRG] Update SourceEstimate.plot() API

### PR DESCRIPTION
**Pull request by christianbrodbeck**
Addresses https://github.com/mne-tools/mne-python/issues/3351: implement changes in the PySurfer API 

- replace use of `config_opts` (deprecated by pysurfer) with corresponding arguments
- add `time_unit` parameter to deprecate conversion of time to ms.

### CODE REVIEW

**Comment by christianbrodbeck** [comment]
I get the error below, the relevant code is [here](https://github.com/mne-tools/mne-python/pull/3375/files#diff-3623e826cb12cb2dfa80dddf3b746d02R1397), can two parameters not be described in one entry?

```
FAIL: Test module docsting formatting
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/mne-tools/mne-python/mne/utils.py", line 733, in dec
    return function(*args, **kwargs)
  File "/home/travis/build/mne-tools/mne-python/mne/tests/test_docstring_parameters.py", line 136, in test_docstring_parameters
    raise AssertionError(msg)
AssertionError: 
mne.source_estimate.SourceEstimate.plot arg mismatch: ['background', 'background, foreground', 'foreground']
mne.viz._3d.plot_source_estimates arg mismatch: ['background', 'background, foreground', 'foreground']
>>  raise AssertionError("\nmne.source_estimate.SourceEstimate.plot arg mismatch: ['background', 'background, foreground', 'foreground']\nmne.viz._3d.plot_source_estimates arg mismatch: ['background', 'background, foreground', 'foreground']")
```

**Comment by Eric89GXL** [comment]
No, they cannot currently

**Comment by christianbrodbeck** [comment]
Fixed! Thanks @Eric89GXL !

File: mne/source_estimate.py [hunk 1]
```diff
@@ -1395,7 +1419,10 @@ def plot(self, subject=None, surface='inflated', hemi='lh',
                                       config_opts=config_opts,
                                       subjects_dir=subjects_dir, figure=figure,
                                       views=views, colorbar=colorbar,
-                                      clim=clim)
+                                      clim=clim, cortex=cortex, size=size,
+                                      background=background,
+                                      foreground=foreground,
+                                      time_unit=time_unit)
```
Comment (jaeilepp) [inline mne/source_estimate.py]: While we're at it, could there be a param for setting the time point showed (like ``set_data_time_index``). Sure, you can do it after plotting, but it doesn't seem to work too well on some settings.

Comment (christianbrodbeck) [inline mne/source_estimate.py]: Unless we also modify PySurfer, this would have to be done in a call to `Brain.set_time()` anyways...

Comment (Eric89GXL) [inline mne/source_estimate.py]: +1 for modifying PySurfer with some introspection or try block to fall back
to the old method if the new way is not available

Comment (christianbrodbeck) [inline mne/source_estimate.py]: @Eric89GXL do I understand correctly that you are +1 for adding an `initial_time_shown` parameter to `Brain.add_data()`?

Comment (Eric89GXL) [inline mne/source_estimate.py]: Sure, or time_index or some other less verbose name

Comment (christianbrodbeck) [inline mne/source_estimate.py]: Ok, to keep things simpler I would suggest a separate PR

File: mne/viz/_3d.py [hunk 2]
```diff
@@ -598,11 +597,13 @@ def _limits_to_control_points(clim, stc_data, colormap):
 
 
 def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
-                          colormap='auto', time_label='time=%0.2f ms',
+                          colormap='auto', time_label='auto',
```
Comment (Eric89GXL) [inline mne/viz/_3d.py]: why change this one?

File: mne/viz/_3d.py [hunk 3]
```diff
@@ -671,45 +673,74 @@ def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
...
 
+    if LooseVersion(surfer.__version__) < LooseVersion('0.6'):
+        raise ImportError("This function requires PySurfer 0.6 (you are "
+                          "running version %s). You can update PySurfer "
+                          "using:\n\n    $ pip install -U pysurfer" %
+                          surfer.__version__)
+
+    if time_unit is None:
+        warn("The time_unit parameter default will change from 'ms' to 's' "
+             "in MNE 0.14. To avoid this warnign specify the parameter "
```
Comment (Eric89GXL) [inline mne/viz/_3d.py]: warning

File: mne/viz/_3d.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/viz/_3d.py]: ...oh I see why below, never mind

File: mne/viz/_3d.py [hunk 4]
```diff
@@ -733,24 +764,29 @@ def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
...
+                      background, foreground, figure, subjects_dir, views,
+                      config_opts=config_opts)
+
+    if time_unit == 's':
+        times = stc.times
+    elif time_unit == 'ms':
+        times = 1e3 * stc.times
+    else:
+        raise ValueError("time_unit needs to be 's' or 'ms', got %s" %
+                         repr(time_unit))
```
Comment (Eric89GXL) [inline mne/viz/_3d.py]: This check should go toward the top -- checking parameters as early as possible is best, plus I think it can be combined a bit with the other ways that it time units get triaged above

File: mne/viz/_3d.py (hunk 2 above)
Comment (christianbrodbeck) [inline mne/viz/_3d.py]: Because the actual label is `time_label % time`, so without changing it, after plotting `stc.plot(time_unit='s')` the user would see "time=-0.01 ms" when the actual time is -100 ms (-0.1 s)...

**Comment by Eric89GXL** [comment]
LGTM +1 for merge
